src/renate/benchmark/models/l2p.py

```diff
@@ -108,6 +108,123 @@ def forward(self, x: torch.Tensor, manual_prompt_indices: Optional[torch.LongTen
+    """
+
+    def __init__(
+        self,
+        pretrained_model_name_or_path="google/vit-base-patch16-224",
+        image_size: int = 32,
+        patch_size: int = 4,
+        num_layers: int = 12,
+        num_heads: int = 12,
+        hidden_dim: int = 768,
+        mlp_dim: int = 3072,
+        dropout: float = 0.1,
+        attention_dropout: float = 0.1,
+        num_outputs: int = 10,
+        prediction_strategy: Optional[PredictionStrategy] = None,
+        add_icarl_class_means: bool = True,
+    ) -> None:
+        super().__init__()
+        if "vit" in pretrained_model_name_or_path:
+            self.transformer = VisionTransformer(
+                pretrained_model_name_or_path=pretrained_model_name_or_path,
+                image_size=image_size,
+                patch_size=patch_size,
+                num_layers=num_layers,
+                num_heads=num_heads,
+                hidden_dim=hidden_dim,
+                mlp_dim=mlp_dim,
+                dropout=dropout,
+                attention_dropout=attention_dropout,
+                num_outputs=num_outputs,
+                prediction_strategy=prediction_strategy,
+                add_icarl_class_means=add_icarl_class_means,
+            )
+            self._is_text_transformer = False
+        else:
+            self.transformer = HuggingFaceSequenceClassificationTransformer(
+                pretrained_model_name_or_path=pretrained_model_name_or_path,
+                num_outputs=num_outputs,
+                prediction_strategy=prediction_strategy,
+                add_icarl_class_means=add_icarl_class_means,
+            )
+            for named_param, value in self.transformer.named_parameters():
+                if value.shape[0] == self.transformer._backbone.config.vocab_size:
+                    self.word_embeddings = self.transformer.get_submodule(
+                        named_param.replace(".weight", "")
+                    )
+                    break
+
+            self._is_text_transformer = True
+
+        self.transformer._tasks_params.clear()
+        self.transformer.eval()
+        for p in self.transformer.parameters():
+            p.requires_grad_(False)
+
+    def forward(
+        self, x: torch.Tensor, prompt: Optional[torch.Tensor] = None, cls_feat: bool = True
+    ) -> torch.Tensor:
+        """
+        Args:
+            x: Input torch tensor.
+            prompt: Prompt tensor. Defaults to None.
+            cls_feat: Whether to extract [CLS] token or to return full feature tensor.
+                Ignored for text transformer. Defaults to True.
+        """
+        if prompt is None:
+            return (
+                self.transformer.get_features(x)
+                if self._is_text_transformer
+                else self.transformer.get_features(x, cls_feat=cls_feat)
+            )
+            # text transformers dont support cls_feat.
+        else:
+            if self._is_text_transformer:
```

Comment:
we can flatten the if else(if else) part to if, elif, else

Reply:
Fixed in d8ef1c81da1e7e1e714a7ef96ab8beb775f0bad4